### PR TITLE
Gui rewrite.

### PR DESCRIPTION
**Pull request by raziel1000**
- Gui: Bug fixes and cleanup.
- Gui: Performance improvement (loading, resizing...etc)
- Added a simple PKG Viewer(Settings-> Utils -> PKG Viewer), added pkg folders will be saved.
- PKG Viewer: Shows game info(name, id, region...etc)
- PKG Viewer: Right click -> Install PKG to install/extract a game. Patch installation is also possible.
- Added option to dump game list  (Settings -> Utils -> Dump Game List), will be dumped to emu folder GameList.txt

### CODE REVIEW

**Comment by georgemoralis** [comment]
oh ;0 new gui ;0

**Comment by georgemoralis** [comment]
bug: pkg viewer crashes if the pkg folder contains other files . e.g i had some rar files in the same folder

File: src/qt_gui/game_grid_frame.cpp [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+    int rowCount = m_games_.size() / gamesPerRow;
+    if (m_games_.size() % gamesPerRow != 0) {
+        rowCount += 1; // Add an extra row for the remainder
+    }
+    std::vector<int> indices;
+    for (int i = 0; i < m_games_.size(); i++) {
+        indices.push_back(i);
+    }
+    std::vector<std::future<QPixmap>> futures;
+    for (int index : indices) {
```
Comment (SachinVin) [inline src/qt_gui/game_grid_frame.cpp]: wouldn't a simple `for (int index = 0; index < m_games_.size(); index++) {` work?

Comment (raziel1000) [reply src/qt_gui/game_grid_frame.cpp]: Thanks! we can even remove it and move it inside the other loop
Edit: committed now :)

Comment (SachinVin) [reply src/qt_gui/game_grid_frame.cpp]: You're welcome :)
BTW out of curiosity was this function written by AI?

**Pull request merged**